src/test/SkipCallsTestHook.sol

```diff
@@ -163,9 +164,10 @@ contract SkipCallsTestHook is BaseTestHooks, Test {
 
         assert(delta0 < 0 || delta1 < 0);
         assert(!(delta0 > 0 || delta1 > 0));
-
+        manager.sync(key.currency0);
```

Comment:
mm this is ok but do you think that this contract should just inherit from PoolTestBase so that it can access `_settle` and `_take` instead of custom impl?
Can fix that in a different PR if you think thats better

Reply:
i think sara was open to adding this library to core:

https://github.com/saucepoint/v4-periphery/blob/position-manager/contracts/libraries/CurrencySettleTake.sol (needs to be updated to call .sync() before calling .transferFrom())

the library should replace these calls + PoolTestBase's _settle and _take with something thats a bit more syntactic sugary 😸 

---

In any case, I'd like to see a unified way of settling/taking across core + periphery. either with a library or a contract

Reply:
tangentially, the library works better in periphery after PoolManager has been deployed 😭 

you can save the PoolManager address as a constant and go from:

`key.currency0.settle(IPoolManager(manager), address(payer), amount, useClaims)`
to
`key.currency.settle(address(payer), amount, useClaims)`

Reply:
Yeah +1 to @saucepoint's lib. Definitely in favor of adding it. Let's push this to a followup PR. Outlined issue here. @saucepoint can you take this?

Re your second point... could we just have one in core for testing purposes? And then expose that one in periphery for prod use case?